### ISSUE TITLE
[Case Study] Is it possible to have nested worktrees?

### ISSUE THREAD

**Issue by zdm** [yarnpkg/berry#1027]
Hi,
I have following structure.
```
(1) package.json - workspaces: ["*"]
|--- (2) package.json - workspaces: ["*"]
     |--- (3) package.json
```
How it is possible to make (3)package.json to be a part of worktree (2)?

Currently when I run `yarn` from dir (3) it is trying to create .yarn.lock under top-level dir (1).

So, I need to make worktree (2) isolated from worktree (1) but is should be a subdirectory of (1).

I hope, you understand my question. ;-)

**Comment by zdm** [comment]
In other words:
1. I have two projects;
2. Both are configured as workspaces;
3. Project2 if located in subdirectory of project1;

I need to make project2 worktree to be isolated from project1.

**Comment by arcanis** [comment]
If I understand correctly there will always be a single lockfile with the current model. We may consider read-only views in the future, but not until the next major at the very least.

**Comment by eliw00d** [comment]
I am also trying out nested worktrees but the documentation is not very clear on how to do so. I have the following setup:
```
/package.json (workspaces: ["backend", "frontend"])
/backend/package.json (workspaces: ["services/*"])
/backend/services/**/package.json
/frontend/package.json (workspaces: ["applications/*"])
/frontend/applications/**/package.json
```
However, when I install from the root the dependencies in services and applications do not get installed.

**Comment by merceyz** [comment]
This issue is seems to boil down to per workspace lockfiles so closing as a duplicate of https://github.com/yarnpkg/berry/issues/1223

> However, when I install from the root the dependencies in services and applications do not get installed.

@eliw00d That looks correct, could you open a issue with a full reproduction?
